types/fh-sync.d.ts

```diff
@@ -0,0 +1,314 @@
+      /** @type {Number} the concurrency value of the sync workers. Default is 1. Can set to 0 to disable the syncWorker completely. */
+      syncWorkgerConcurrency?: number;
+      /** @type {Object} the backoff strategy for the sync worker to use. 
+      * Default strategy is `exp` (exponential) with a max delay of 1s. The min value will always be the same as `syncWorkerInterval`
+      * Other valid strategies are `none` and `fib` (fibonacci).*/
+      syncWorkerBackoff?: SyncWorkerBackoff;
+      /** @type {Number} how often the scheduler should check the datasetClients, in ms. Default: 500 */
+      schedulerInterval?: number;
+      /** @type {Number} the max time a scheduler can hold the lock for, in ms. Default: 20000 */
+      schedulerLockMaxTime?: number;
+      /** @type {String} the default lock name for the sync scheduler */
+      schedulerLockName?: string;
+      /**@type {Number} the default concurrency value when update dataset clients in the sync API. Default is 10. In most case this value should not need to be changed */
+      datasetClientUpdateConcurrency?: number;
+      /**@type {Boolean} enable/disable collect sync stats to allow query via an endpoint */
+      collectStats?: boolean;
+      /**@type {Number} the number of records to keep in order to compute the stats data. Default is 1000. */
+      statsRecordsToKeep?: number;
+      /**@type {Number} how often the stats should be collected. In milliseconds. */
+      collectStatsInterval?: number;
+      /**@type {String} the host of the influxdb server. If set, the metrics data will be sent to the influxdb server. */
+      metricsInfluxdbHost?: string;
+      /**@type {Number} the port of the influxdb server. It should be a UDP port. */
+      metricsInfluxdbPort?: number;
+      /**@type {Number} the concurrency value for the component metrics. Default is 10. This value should be increased if there are many concurrent workers. Otherwise the memory useage of the app could go up.*/
+      metricsReportConcurrency?: number;
+      /**@type {Boolean} if cache the dataset client records using redis. This can help improve performance for the syncRecords API.
+      * Can be turned on if there are no records are shared between many different dataset clients. Default is false.*/
+      useCache?: boolean;
+      /**@type {Number} the TTL (Time To Live) value for the messages on the queue. In seconds. Default to 24 hours. */
+      queueMessagesTTL?: string;
+      /**@type {String} specify the maximum retention time of an inactive datasetClient. Any inactive datasetClient that is older than this period of time will be removed.*/
+      datasetClientCleanerRetentionPeriod?: string;
+      /** @type {String} specify the frequency the datasetClient cleaner should run. Default every hour ('1h').*/
+      datasetClientCleanerCheckFrequency?: string;
+    }
+
+    /**
+     * Backoff Strategy
+     * Example: {strategy: 'exp', max: 60*1000},
```

Comment:
The type for `max` for PendingWorkerBackoff and AckWorderBackoff are strings, but I think they need to be numbers.